### ISSUE TITLE
examples in README should have types

### ISSUE THREAD

**Issue by cdepillabout** [aktowns/purescript-simple-dom#48]
Coming from Haskell, it's hard to follow the examples in the `README` without type information.  Putting type annotations on the `setContents` function, `modifyLinkTarget` function, etc. would make the example much easier to follow.

**Comment by pedrofurla** [comment]
+1

**Comment by pierrebeaucamp** [comment]
On a similar note, the examples are a bit outdated. 
For example, there is no `addEventListener` method anymore (I think it was replaced with `unsafeAddEventListener`). 
It would be great to have some more documentation and some working examples for this library.

**Comment by cdepillabout** [comment]
@pierrebeaucamp The [`purescript-dom`](https://github.com/purescript-contrib/purescript-dom) library was created because this library doesn't seem to being updated any more.  `purescript-dom` is also somewhat blessed by the community.

If you're looking for a DOM manipulation library, you may want to check it out.

**Comment by pierrebeaucamp** [comment]
@cdepillabout I know about `purescript-dom` but it's stated in its `README.md`:
> The API isn't primarily intended for "human consumption" [...]

I find it simpler to use `purescript-simple-dom` for manual DOM manipulation. I think there is definitely enough use cases for both libraries to exist side by side. Ideally, `purescript-simple-dom` should make use of `purescript-dom`.

**Comment by cdepillabout** [comment]
> I know about purescript-dom but it's stated in its README.md:
>> The API isn't primarily intended for "human consumption" [...]

Maybe this used to be the case, but it's not true anymore.  purescript-dom is currently the "blessed" library if you need do to dom operations (and you can't use a library like thermite or halogen).

> Ideally, purescript-simple-dom should make use of purescript-dom.

This is definitely true though.
